lib/src/Uncertainty/Distribution/RandomMixture.cxx

```diff
@@ -453,6 +454,12 @@ void RandomMixture::setDistributionCollectionAndWeights(const DistributionCollec
       // Aggregate the atoms
       atomCandidates.add(mixture->getDistributionCollection());
     } // atom is a RandomMixture
+    else if (atomKind == "TruncatedDistribution")
+    {
+      const TruncatedDistribution * truncatedDistribution(static_cast< const TruncatedDistribution * >(atom.getImplementation().get()));
```

Comment:
here too

Reply:
I checked the sources, and we have static_cast conversions of pointers of complex types in several othr places (HMatrixImplementation, NLopt, MemoizeFunction) in addition to the numerous static_cast of simple types. Do you think we should change all of them? Is the penalty involved in using dynamic_cast instead of static_cast that important? My main point is that I hope we can rely on the information given by getClassName() to perform a cast, it is exactly why it has been introduced in the first place. Not using it is like wearing a condom and have no sex: something which does not carry a good confidence of developers into their own developement...

Reply:
Imho it's just a good practice, nothing special about your code. You dont have to replace it everywhere.